Makefile

```diff
@@ -1,4 +1,4 @@
-all: install_requirements configure_host launch_mgmt_cluster verify
+all: clean install_requirements configure_host launch_mgmt_cluster verify
```

Comment:
I don't think this fits typical makefile usage, you'd always do e.g `make clean all` and just `make` could be run non-destructively in the case where something fails during the `make` process?

Reply:
agreed - I'd suggest trying to detect an unclean environment in one of the scripts and giving a helpful message.

Reply:
@hardys the first approach was adding in the documentation to run make clean before consecutive make in case of error
https://github.com/metal3-io/metal3-io.github.io/pull/93/files

@russellb I did think about the script and helpful message but the script to detect error and giving helpful message would be more efficient if we would have more than one message to display, in our case it is always make clean if the we get error with make?

Reply:
My point is that as a developer you might in some cases not want to `make clean` in the case of error.

Also for anyone who's used `make` in the context of software development this behaviour is probably unexpected, so personally I'd prefer to just document running `make clean` explicitly.

Reply:
What is the specific problem you're trying to solve here, is there a script which fails when re-running?  Can we fix that so it's idempotent perhaps?